metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy

```diff
@@ -187,7 +176,7 @@ class FilterExperimentData implements FilterExperimentDataInput {
             if (file.contains(prepID.toString()) || file.contains(sampleName.toString())) {
                 valid = true
             } else {
-                LOG.warn "File of sample $prepID does not follow the naming conventions: $file"
+                messageOutputPort.invokeError("File of sample $prepID does not follow the naming conventions: $file",FilterExperimentData.class)
```

Comment:
you would throw an exception here and there is no need to interact with the messageOutputPort

Reply:
Same as for the filename validation, if I thrown an exception the program execution will stop which is not desired. Therefore I transfer this information through the output port to handle the message.

Reply:
You can catch the exception and log. When you isolate the logical comparison into a method and call this method with a try catch block that catches an exception for each filename, fills empty and logs the error you are good.

Reply:
#7